### ISSUE TITLE
ESM packages support Swiper.js 7

### ISSUE THREAD

**Issue by flean** [meteor/meteor#11631]
Right now I'm stuck with version 6 of swiper and would like to upgrade to 7 but it requires ESM support. 

Is there a way for us to use ESM packages in meteor?

[SwiperJS](https://swiperjs.com/migration-guide)

**Comment by StorytellerCZ** [comment]
I don't follow, are you getting an error when you are trying to import from Swipper 7?

For example:
```
import Swiper from 'swiper/swiper-bundle.esm.js';
```
or
```
import Swiper from 'swiper';
```

**Comment by flean** [comment]
```
// Import Swiper React components
import { Swiper, SwiperSlide } from 'swiper/react';

// Import Swiper styles
import 'swiper/css';

```

**Comment by StorytellerCZ** [comment]
Shouldn't it be?
```js
import Swiper, { SwiperSlide } from 'swiper/react';
```

**Comment by flean** [comment]
No, that comes directly from their docs and the react component inherits swiper core. At least thats my understanding.

**Comment by StorytellerCZ** [comment]
I still don't understand what the problem here is.

**Comment by flean** [comment]
It can’t find the module, app breaks. I have to use version 6 for it to work. It looks like meteor can’t use ESM packages. I’ll setup a meteor app only loading that component.

**Comment by flean** [comment]
here is the [repo](https://github.com/flean/simple-swiper-example) with version 7

Server
```
Unable to resolve some modules:

  "swiper/css" in /Users/reynaldoreyna/Documents/Development/Meteor.nosync/learning/simple-swiper-example/imports/ui/App.jsx (web.browser)

If you notice problems related to these missing modules, consider running:

  meteor npm install --save swiper
```
 Client Side
```
[Log] HMR: connected (hot-module-replacement.js, line 330)
[Error] Error: Cannot find module 'swiper/react'
	(anonymous function) (modules-runtime-hot.js:253)
	moduleLink (modules.js:300)
	module (app.js:55)
	fileEvaluate (modules-runtime-hot.js:388)
	require (modules-runtime-hot.js:270)
	moduleLink (modules.js:307)
	module (app.js:218)
	fileEvaluate (modules-runtime-hot.js:388)
	require (modules-runtime-hot.js:270)
	Global Code (app.js:240)
```

**Comment by StorytellerCZ** [comment]
OK, I see now!

Yeah, that is indeed not working. Right now workaround would be to import from the files directly like this:
```js
import 'swiper/swiper.min.css';
```

But that is only a stop gap solution.

@zodern @renanccastro @filipenevola 
The issue here is that in `package.json` the exports are defined under `exports` prop, in this case like this:
```json
"exports": {
    ".": "./swiper.esm.js",
    "./core": "./swiper.esm.js",
    "./bundle": "./swiper-bundle.esm.js",
    "./css": "./swiper.min.css",
    "./css/bundle": "./swiper-bundle.min.css",
    "./css/a11y": "./modules/a11y/a11y.min.css",
    "./css/autoplay": "./modules/autoplay/autoplay.min.css",
...
```

**Comment by trusktr** [comment]
Even TypeScript barely [_just_ added](https://github.com/microsoft/TypeScript/issues/33079#issuecomment-926936955) support for recognizing package.json `exports` fields.

The only solution right now is to import the files directly until Meteor adds support for it as well.

Keep one thing in mind about this workaround though: importing files directly will work in Meteor, but will fail in Node apps, or in Webpack which also follows the latest Node ESM spec. If you have some code in package A, for example, and it imports files directly from some package B, this will work in Meteor, but it will throw an error in Node or Webpack apps.

**Comment by trusktr** [comment]
This is a duplicate of https://github.com/meteor/meteor/issues/10906

**Comment by StorytellerCZ** [comment]
Some notes from me digging around this:
Babel plans to become ESM package in Babel 8: https://babeljs.io/docs/en/roadmap#babel-8
There is this esm package which should make things work: https://www.npmjs.com/package/esm
But Node should supporting this already: https://nodejs.org/docs/latest-v14.x/api/esm.html
